crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -300,36 +324,24 @@ async fn graphql_handler(
     result.into()
 }
 
-async fn health_checks(
-    ConnectInfo(addr): ConnectInfo<SocketAddr>,
-    schema: axum::Extension<SuiGraphQLSchema>,
-    req: GraphQLRequest,
-) -> impl axum::response::IntoResponse {
-    // Simple request to check if the DB is up
-    // TODO: add more checks and figure out better ways to do these health checks
-    let mut req = req.into_inner();
-    req.data.insert(addr);
-    // insert the NIL UUID which helps avoid logging these health requests
-    req.data.insert(HEALTH_UUID);
-    req.query = r#"
-        query {
-            chainIdentifier
+/// Connect via a TCPStream to the DB to check if it is alive
+async fn health_checks(State(connection): State<ConnectionConfig>) -> StatusCode {
+    let db_url = connection.db_url;
+    let url = reqwest::Url::parse(db_url.as_str());
+    if let Ok(url) = url {
+        if let (Some(h), Some(p)) = (url.host_str(), url.port()) {
+            let listener = TcpStream::connect(format!("{}:{}", h, p));
+            if listener.is_err() {
+                StatusCode::INTERNAL_SERVER_ERROR
+            } else {
+                StatusCode::OK
+            }
+        } else {
+            StatusCode::INTERNAL_SERVER_ERROR
         }
-        "#
-    .to_string();
-    let db_up = match schema.execute(req).await.is_ok() {
-        true => "UP",
-        false => "DOWN",
-    };
-    let uptime = get_or_init_server_start_time()
-        .await
-        .elapsed()
-        .as_secs_f64();
-    format!(
-        r#"{{"status": "UP","uptime": {},"checks": {{"DB": "{}",}}}}
-        "#,
-        uptime, db_up
-    )
+    } else {
+        StatusCode::INTERNAL_SERVER_ERROR
+    }
```

Comment:
nit: Some minor tidying is possible here, using early return style, and supporting optional ports:

```rust
async fn health_checks(State(config): State<ConnectionConfig>) -> StatusCode {
    let Ok(url) = reqwest::Url::parse(config.db_url.as_str()) else {
        return StatusCode::INTERNAL_SERVER_ERROR;
    };

    let Some(host) = url.host_str() else {
        return StatusCode::INTERNAL_SERVER_ERROR;
    };

    let tcp_url = if let Some(port) = url.port() {
        format!("{host}:{port}")
    } else {
        host.to_string();
    };

    if TcpStream::connect(tcp_url).is_err() {
        StatusCode::INTERNAL_SERVER_ERROR
    } else {
        StatusCode::OK
    }
}
```